File: scala/src/test/scala/com/laamella/code_state_machine/builder/ScxmlParserTest.scala
package com.laamella.code_state_machine.builder

import com.laamella.code_state_machine.action.LogAction
import com.laamella.code_state_machine.io.DotOutput
import com.laamella.code_state_machine.priority.AutomaticPriority
import com.laamella.code_state_machine.{AlwaysCondition, Condition, UnitSpec}
import org.xml.sax.InputSource

class ScxmlParserTest extends UnitSpec {

  behavior of "the SCXML parser"

  it should "correctly read an SCXML file" in {
    val inputSource = new InputSource(classOf[ScxmlStateMachineBuilder[String, String]].getResourceAsStream("/test.scxml"))
    val scxmlParser = new ScxmlStateMachineBuilder[String, String](inputSource) {
      override def interpretStateName(name: String): String = name

      override def interpretEvent(attribute: String): () => Unit = new LogAction(attribute)

      override def interpretCondition(attribute: String): Condition[String] = new AlwaysCondition[String]()
    }
    val stateMachine = scxmlParser.build()
    System.out.println(new DotOutput[String, String, AutomaticPriority]().getOutput(stateMachine))
//    val stateMachine = builder.build()
    // TODO finish tests
  }
}

File: java/src/test/java/com/laamella/code_state_machine/builder/ScxmlParserTest.java
package com.laamella.code_state_machine.builder;

import java.io.IOException;

import javax.xml.parsers.ParserConfigurationException;

import org.junit.Test;
import org.xml.sax.InputSource;
import org.xml.sax.SAXException;

import com.laamella.code_state_machine.Action;
import com.laamella.code_state_machine.Condition;
import com.laamella.code_state_machine.StateMachine;
import com.laamella.code_state_machine.action.LogAction;
import com.laamella.code_state_machine.condition.AlwaysCondition;
import com.laamella.code_state_machine.io.DotOutput;

public class ScxmlParserTest {
    @Test
    public void readTestScxml() throws ParserConfigurationException, SAXException, IOException {
        final InputSource inputSource = new InputSource(ScxmlStateMachineBuilder.class.getResourceAsStream("/test.scxml"));
        final ScxmlStateMachineBuilder<String, String> scxmlParser = new ScxmlStateMachineBuilder<>(inputSource) {
            @Override
            protected String interpretStateName(final String name) {
                return name;
            }

            @Override
            protected Action interpretEvent(final String attribute) {
                return new LogAction(attribute);
            }

            @Override
            protected Condition<String> interpretCondition(final String attribute) {
                return new AlwaysCondition<>();
            }
        };
        final StateMachine<String, String, Integer> stateMachine = scxmlParser.build();
        System.out.println(new DotOutput<String, String, Integer>().getOutput(stateMachine));
        //		final StateMachine<String, String, Priority> stateMachine = builder.build();
        // TODO finish tests
    }
}
